test/libp2p/multi/utils/uvarint_test.cpp

```diff
@@ -61,3 +61,43 @@ TEST(UVarint, CalculateSize) {
   uint8_t another_bytes[] = {0x71, 0xA3, 0x75, 0x43, 0xAA};
   ASSERT_EQ(UVarint::calculateSize(gsl::span(another_bytes, 5)), 1);
 }
+
+/**
+ * @given Sample integers from (N/2, N)
+ * @when Encoding and decoding back sample integer
+ * @then Encoding/decoding must be successful without losses of data
+ */
+TEST(UVarint, ReversibilitySuccess) {
+  uint64_t max = std::numeric_limits<uint64_t>::max() / 2;
+  for (uint64_t data = 2; data < max; data *= 2) {
+    UVarint var{data};
+    uint64_t decoded = var.toUInt64();
+    ASSERT_EQ(data, decoded);
+  }
+}
+
+/**
+ * @given Minimum and maximum possible values
+ * @when Encoding and decoding back sample integer
+ * @then Encoding/decoding must be successful without losses of data
+ */
+TEST(UVarint, EncodeLimitsSuccess) {
```

Comment:
```suggestion
TEST(UVarint, EncodeLimitsAreCorrect) {
```
I guess it can be named not only success or failure, specially when there is not failure nor success.
This is my suggestion.